docs/USAGE-GUIDE.md

```diff
@@ -1706,23 +1706,29 @@ this will allow for gradual and linear transition of a key from encrypt-decrypt
 
 ### Asynchronous private key operations related calls
 
-When s2n-tls is used in non-blocking mode, this set of functions allows user
+When s2n-tls is used in non-blocking mode, this set of functions allows users
 to move execution of CPU-heavy private key operations out of the main
-event loop, preventing **s2n_negotiate** blocking the loop for a few
+event loop, preventing **s2n_negotiate** from blocking the loop for a few
 milliseconds each time the private key operation needs to be performed.
```

Comment:
is it worth mentioning other possible use cases, like Hardware Security Modules (ex: PKCS#11, TPM, etc) where the private key is hidden and an external library must perform these operations?